iroh-util/src/lock.rs

```diff
@@ -28,52 +35,143 @@ impl ProgramLock {
 
         // Even if we manage to lock the file this won't last since the drop implementation
         // of FileGuard releases the underlying lock.
-        if let Ok(file) = File::create(&self.path) {
+        if let Ok(file) = File::open(&self.path) {
             file_guard::try_lock(&file, Lock::Exclusive, 0, 1).is_err()
         } else {
             false
         }
     }
 
     /// Try to acquire a lock for this program.
-    pub fn acquire(&mut self) -> Result<()> {
-        let file = Rc::new(File::create(&self.path)?);
+    pub fn acquire(&mut self) -> AnyhowResult<()> {
+        // ensure path to lock exists
+        std::fs::create_dir_all(&crate::iroh_data_root()?)?;
+        let mut file = OpenOptions::new()
+            .write(true)
+            .create_new(true)
+            .open(&self.path)?;
+        file.write_all(process::id().to_string().as_bytes())?;
+        let file = Rc::new(file);
 
         file_guard::lock(file, Lock::Exclusive, 0, 1)
             .map(|lock| self.lock = Some(lock))
             .map_err(|err| err.into())
     }
 }
 
+/// Attempt to remove a stray lock file that wasn't cleaned up, returns true
+/// if a lock is successfully deleted, and will only attempt to delete if the
+/// lock is not currently held
+pub fn try_cleanup_dead_lock(prog_name: &str) -> AnyhowResult<bool> {
+    let lock = ProgramLock {
+        path: crate::iroh_data_path(&format!("{}.lock", prog_name))?,
+        lock: None,
+    };
+    if lock.is_locked() {
+        info!("lock {} is currently active, cannot remove", prog_name);
+        return Ok(false);
+    }
+    match std::fs::remove_file(lock.path) {
+        Err(e) => {
+            info!("error removing {} lockfile: {}", prog_name, e);
+            Err(anyhow!("removing dead lockfile: {}", e))
```

Comment:
I think seeing something like "Removing dead lockfile" as an error message may be a bit confusing, depending on how it's rendered. I suggest "could not remove dead lockfile: "